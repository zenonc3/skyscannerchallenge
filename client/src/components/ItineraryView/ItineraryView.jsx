import PropTypes from 'prop-types';
import React from 'react';
import Itinerary from './../Itinerary';


const ItineraryView = ({ itineraries }) => {
  const itineraryCards = [];
  itineraries.forEach((itinerary) => {
    itineraryCards.push(<Itinerary key={itinerary.pricing.bookingUrl} {...itinerary} />);
  });
  return itineraryCards.length > 100 ? itineraryCards.slice(0, 100) : itineraryCards;
};

ItineraryView.propTypes = {
  itineraries: PropTypes.arrayOf(Itinerary).isRequired,
};

export default ItineraryView;
