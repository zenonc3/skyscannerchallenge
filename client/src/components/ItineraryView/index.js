import ItineraryView from './ItineraryView';

export default ItineraryView;
