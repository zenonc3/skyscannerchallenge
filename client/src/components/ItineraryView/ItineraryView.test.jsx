import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ItineraryView from './ItineraryView';


describe('Itinerary', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ItineraryView itineraries={[]} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

