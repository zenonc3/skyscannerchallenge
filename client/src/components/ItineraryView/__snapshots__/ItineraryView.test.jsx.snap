// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itinerary should render correctly 1`] = `Array []`;
